src/main/webapp/app/quiz/edit/short-answer-question/edit-short-answer-question.component.ts

```diff
@@ -568,4 +584,115 @@ export class EditShortAnswerQuestionComponent implements OnInit, OnChanges, Afte
+        this.resetQuestionTitle();
+        this.question.invalid = this.backupQuestion.invalid;
+        this.question.randomizeOrder = this.backupQuestion.randomizeOrder;
+        this.question.scoringType = this.backupQuestion.scoringType;
+        this.question.solutions = JSON.parse(JSON.stringify(this.backupQuestion.solutions));
+        this.question.correctMappings = JSON.parse(JSON.stringify(this.backupQuestion.correctMappings));
+        this.question.spots = JSON.parse(JSON.stringify(this.backupQuestion.spots));
+        this.resetQuestionText();
+    }
+
+    /**
+     * @function resetSpot
+     * @desc Resets the spot
+     * @param spot {spot} the spot, which will be reset
+     */
+    resetSpot(spot: ShortAnswerSpot): void {
+        // Find matching spot in backupQuestion
+        const backupSpot = this.backupQuestion.spots.find(currentSpot => currentSpot.id === spot.id);
+        // Find current index of our spot
+        const spotIndex = this.question.spots.indexOf(spot);
+        // Remove current spot at given index and insert the backup at the same position
+        this.question.spots.splice(spotIndex, 1);
+        this.question.spots.splice(spotIndex, 0, backupSpot);
+    }
+
+    /**
+     * @function deleteSpot
+     * @desc Delete the given spot
+     * @param spotToDelete {object} the spot to delete
+     */
+    deleteSpot(spotToDelete: ShortAnswerSpot): void {
+        this.question.spots = this.question.spots.filter(spot => spot !== spotToDelete);
+        this.deleteMappingsForSpot(spotToDelete);
+
+        this.textParts = this.question.text.split(/\n+/g).map(t => t.split(/\s+(?![^[]]*])/g));
+        this.textParts = this.textParts.map(part => part.filter(text => text !== '[-spot ' + spotToDelete.spotNr + ']'));
+
+        this.question.text = this.textParts.map(textPart => textPart.join(' ')).join('\n');
+    }
+
+    /**
+     * @function deleteMappingsForSpot
+     * @desc Delete all mappings for the given spot
+     * @param spot {object} the spot for which we want to delete all mappings
+     */
+    deleteMappingsForSpot(spot: ShortAnswerSpot): void {
+        if (!this.question.correctMappings) {
+            this.question.correctMappings = [];
+        }
+        this.question.correctMappings = this.question.correctMappings.filter(
+            mapping => !this.shortAnswerQuestionUtil.isSameSpot(mapping.spot, spot)
+        );
+    }
+
+    /**
+     * @function setQuestionText
+     * @desc sets the new text as question.text and updates the UI (through textParts)
+     * @param id
+     */
+    setQuestionText(id: string): void {
+        const rowColumn: string [] = id.split('-').slice(1);
+        this.textParts[rowColumn[0]][rowColumn[1]] = (<HTMLInputElement>document.getElementById(id)).value;
+        this.question.text = this.textParts.map(textPart => textPart.join(' ')).join('\n');
+        this.textParts = this.question.text.split(/\n+/g).map(t => t.split(/\s+(?![^[]]*])/g));
```

Comment:
use constants for regex expressions and make sure to comment them!

Reply:
Also consider to extract the splitting and merging into its own methods to make it easier to reuse the functionality from different places

Reply:
Good idea! I already am doing this in the improvements branch, as I need this for participate and the statistics as well. 

It will be integrated in the new branch.